dot/sync/disjoint_block_set_test.go

```diff
@@ -0,0 +1,313 @@
+		want             []*pendingBlock
+	}{
+		{
+			name: "no blocks",
+			want: []*pendingBlock{},
+		},
+		{
+			name: "base case",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					{}: testBlock,
+				},
+			},
+			want: []*pendingBlock{testBlock},
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			s := &disjointBlockSet{
+				blocks: tt.disjointBlockSet.blocks,
+			}
+			blocks := s.getBlocks()
+			assert.Equal(t, tt.want, blocks)
+		})
+	}
+}
+
+func Test_disjointBlockSet_removeLowerBlocks(t *testing.T) {
+	t.Parallel()
+
+	testBlock1 := &pendingBlock{
+		hash:   common.Hash{1},
+		number: 1,
+	}
+	testBlock10 := &pendingBlock{
+		hash:   common.Hash{10},
+		number: 10,
+	}
+
+	tests := []struct {
+		name             string
+		disjointBlockSet disjointBlockSet
+		num              uint
+		remaining        int
+	}{
+		{
+			name: "number 0",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					{1}:  testBlock1,
+					{10}: testBlock10,
+				},
+			},
+			num:       0,
+			remaining: 2,
+		},
+		{
+			name: "number 1",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					{1}:  testBlock1,
+					{10}: testBlock10,
+				},
+			},
+			num:       1,
+			remaining: 1,
+		},
+		{
+			name: "number 11",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					{1}:  testBlock1,
+					{10}: testBlock10,
+				},
+			},
+			num:       11,
+			remaining: 0,
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tt.disjointBlockSet.removeLowerBlocks(tt.num)
+			assert.Equal(t, tt.remaining, len(tt.disjointBlockSet.blocks))
+		})
+	}
+}
+
+func Test_disjointBlockSet_size(t *testing.T) {
+	t.Parallel()
+
+	testBlock1 := &pendingBlock{
+		hash:   common.Hash{1},
+		number: 1,
+	}
+	testBlock10 := &pendingBlock{
+		hash:   common.Hash{10},
+		number: 10,
+	}
+	tests := []struct {
+		name             string
+		disjointBlockSet disjointBlockSet
+		want             int
+	}{
+		{
+			name: "expect 0",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{},
+			},
+			want: 0,
+		},
+		{
+			name: "expect 1",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					testBlock1.hash: testBlock1,
+				},
+			},
+			want: 1,
+		},
+		{
+			name: "expect 2",
+			disjointBlockSet: disjointBlockSet{
+				blocks: map[common.Hash]*pendingBlock{
+					testBlock1.hash:  testBlock1,
+					testBlock10.hash: testBlock10,
+				},
+			},
+			want: 2,
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			s := &disjointBlockSet{
+				blocks: tt.disjointBlockSet.blocks,
+			}
+			assert.Equal(t, tt.want, s.size())
```

Comment:
supernit split the function we are testing away from the assertion, my brain froze and was wondering what we were testing here :smile: 

```suggestion
			size := s.size()
			assert.Equal(t, tt.want, size)
```

Reply:
Updated.